token/jwt/token_test.go

```diff
@@ -0,0 +1,379 @@
+			keyfunc:     defaultKeyFunc,
+			claims:      MapClaims{"foo": "bar"},
+			valid:       false,
+			errors:      ValidationErrorSignatureInvalid,
+		},
+		{
+			name:        "basic nokeyfunc",
+			tokenString: "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJmb28iOiJiYXIifQ.FhkiHkoESI_cG3NPigFrxEk9Z60_oXrOT2vGm9Pn6RDgYNovYORQmmA0zs1AoAOf09ly2Nx2YAg6ABqAYga1AcMFkJljwxTT5fYphTuqpWdy4BELeSYJx5Ty2gmr8e7RonuUztrdD5WfPqLKMm1Ozp_T6zALpRmwTIW0QPnaBXaQD90FplAg46Iy1UlDKr-Eupy0i5SLch5Q-p2ZpaL_5fnTIUDlxC3pWhJTyx_71qDI-mAA_5lE_VdroOeflG56sSmDxopPEG3bFlSu1eowyBfxtu0_CuVd-M42RU75Zc4Gsj6uV77MBtbMrf4_7M_NUTSgoIF3fRqxrj0NzihIBg",
+			keyfunc:     nilKeyFunc,
+			claims:      MapClaims{"foo": "bar"},
+			valid:       false,
+			errors:      ValidationErrorUnverifiable,
+		},
+		{
+			name:        "basic nokey",
+			tokenString: "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJmb28iOiJiYXIifQ.FhkiHkoESI_cG3NPigFrxEk9Z60_oXrOT2vGm9Pn6RDgYNovYORQmmA0zs1AoAOf09ly2Nx2YAg6ABqAYga1AcMFkJljwxTT5fYphTuqpWdy4BELeSYJx5Ty2gmr8e7RonuUztrdD5WfPqLKMm1Ozp_T6zALpRmwTIW0QPnaBXaQD90FplAg46Iy1UlDKr-Eupy0i5SLch5Q-p2ZpaL_5fnTIUDlxC3pWhJTyx_71qDI-mAA_5lE_VdroOeflG56sSmDxopPEG3bFlSu1eowyBfxtu0_CuVd-M42RU75Zc4Gsj6uV77MBtbMrf4_7M_NUTSgoIF3fRqxrj0NzihIBg",
+			keyfunc:     emptyKeyFunc,
+			claims:      MapClaims{"foo": "bar"},
+			valid:       false,
+			errors:      ValidationErrorSignatureInvalid,
+		},
+		{
+			name:        "basic errorkey",
+			tokenString: "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJmb28iOiJiYXIifQ.FhkiHkoESI_cG3NPigFrxEk9Z60_oXrOT2vGm9Pn6RDgYNovYORQmmA0zs1AoAOf09ly2Nx2YAg6ABqAYga1AcMFkJljwxTT5fYphTuqpWdy4BELeSYJx5Ty2gmr8e7RonuUztrdD5WfPqLKMm1Ozp_T6zALpRmwTIW0QPnaBXaQD90FplAg46Iy1UlDKr-Eupy0i5SLch5Q-p2ZpaL_5fnTIUDlxC3pWhJTyx_71qDI-mAA_5lE_VdroOeflG56sSmDxopPEG3bFlSu1eowyBfxtu0_CuVd-M42RU75Zc4Gsj6uV77MBtbMrf4_7M_NUTSgoIF3fRqxrj0NzihIBg",
+			keyfunc:     errorKeyFunc,
+			claims:      MapClaims{"foo": "bar"},
+			valid:       false,
+			errors:      ValidationErrorUnverifiable,
+		},
+		{
+			name:        "valid signing method",
+			tokenString: "",
+			keyfunc:     defaultKeyFunc,
+			claims:      MapClaims{"foo": "bar"},
+			valid:       true,
+			errors:      0,
+		},
+		{
+			name:        "invalid",
+			tokenString: "foo_invalid_token",
+			keyfunc:     defaultKeyFunc,
+			claims:      MapClaims(nil),
+			valid:       false,
+			errors:      ValidationErrorMalformed,
+		},
+		{
+			name:        "valid format invalid content",
+			tokenString: "foo.bar.baz",
+			keyfunc:     defaultKeyFunc,
+			claims:      MapClaims(nil),
+			valid:       false,
+			errors:      ValidationErrorMalformed,
+		},
+		{
+			name:        "wrong key, expected RSA got ECDSA",
```

Comment:
This appears to be

```suggestion
			name:        "wrong key, expected ECDSA got RSA",
```

Reply:
Good catch, fixed, I structured a bit better the tests to make it more explicit what is given and what is expected, I copied tests from `dgrijalva/jwt-go` were a bit confusing.